# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../boomerang/progpath.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- boomerang/loader/BinaryFileFactory.cpp.vanilla	2000-01-01 00:00:00.000000000 +0000
+++ boomerang/loader/BinaryFileFactory.cpp	2006-07-24 18:10:51.000000000 +0000
@@ -105,7 +105,7 @@
 	
 // Load the specific loader library
 #ifndef _WIN32		// Cygwin, Unix/Linux
-	libName = std::string("lib/lib") + libName;
+	libName = std::string("/usr/" "lib/lib") + libName;
 #ifdef	__CYGWIN__
 	libName += ".dll";		// Cygwin wants .dll, but is otherwise like Unix
 #else
--- boomerang/boomerang.cpp.vanilla	2000-01-01 00:00:00.000000000 +0000
+++ boomerang/boomerang.cpp	2006-07-24 17:40:50.000000000 +0000
@@ -893,8 +893,13 @@
 				break;
 			case 'P':
 				progPath = argv[++i];
+#ifdef _MSC_VER
 				if (progPath[progPath.length()-1] != '\\')
 					progPath += "\\";
+#else
+				if (progPath[progPath.length()-1] != '/')
+					progPath += "/";
+#endif
 				break;
 			case 'a':
 				assumeABI = true;
